internal/worker/rundynamic.go

```diff
@@ -230,18 +126,50 @@ func RunDynamicAnalysis(ctx context.Context, pkg *pkgmanager.Pkg, sbOpts []sandb
 
 	LogDynamicAnalysisResult(ctx, pkg, result.LastRunPhase, result.LastStatus)
 
-	if !codeExecutionEnabled {
-		// nothing more to do
-		return result, nil
+	return result, nil
+}
+
+func runDynamicAnalysisPhase(ctx context.Context, pkg *pkgmanager.Pkg, sb sandbox.Sandbox, analysisCmd string, phase analysisrun.DynamicPhase, result *DynamicAnalysisResult) error {
+	if phase == analysisrun.DynamicPhaseExecute {
+		// silently skip execute phase if disabled globally or for specific ecosystem
+		if !isExecutePhaseEnabled(pkg.Ecosystem()) {
+			return nil
+		}
```

Comment:
This logic could be avoided by wrapping `analysisrun.DefaultDynamicPhases()` in here and having it return the correct set of phases based on the `pkg.Ecosystem` and the feature flag.

Reply:
Done